### DISCUSSION TITLE
How to convert the value of AuxVariable to [Function] block?

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27271] [Q&A General] [answered]
Hi all,

I would like to convert AuxVariable value to a function, and then use this function in [Pressure] block of SolidMechanics Module. But I cannot find any type of function to do this. 
Could anyone kindly help me? Thanks!

**Comment by maxnezdyur** [comment, marked as answer]
I think you can use the https://mooseframework.inl.gov/syntax/BCs/CoupledPressure/ action instead. This uses the AuxVariable directly.

**Reply by SomnusYu** [reply]
Thanks!

**Reply by GiudGiud** [reply]
Yeah functions cannot depend on variables. They should only depend on space and time
